src/components/common/TagSelector.tsx

```diff
@@ -0,0 +1,118 @@
+import React, {useState, useEffect, useRef, useCallback, useMemo} from 'react';
+import {
+  TouchableOpacity,
+  FlatList,
+  ScrollView,
+  Keyboard,
+  TouchableWithoutFeedback,
+} from 'react-native';
+import {TagType} from '@/types/TagType';
+import {getTagHistory} from '@apis/Tag';
+import S from './TagSelector.style';
+
+const TagSelector = () => {
```

Comment:
```suggestion
type Props = {
  value: string[];
  options: string[];
  onChange: (val: string) => void;
};

const DropDownSelector = ({
  value,
  options,
  onChange,
} : Props) => {
```

```suggestion
const DropDownSelector = () => {
```

해당 컴포넌트 options 중에서 onChange에 의해서 value를 선택하거나, 새로운 value를 option에 추가하는 기능을 제공하는 컴포넌트이어야 될 것 같아요.

- value: 현재 선택 완료된 값
- options: 선택 가능한 값
- onChange: value가 변화하면 실행될 함수

1. fetchTags 는 해당 컴포넌트의 소비자(`가게 관리 페이지`가 될 것 같네요.)에서 얻은 후 options props를 통해서 건네주면 좋겠습니다.
2. `TagSelector`라는 컴포넌트 이름 대신 드랍다운으로 선택한다는 범용적인 컴포넌트가 되면 좋겠습니다.